### PR TITLE
Add ParsimoniousStepper

### PR DESCRIPTION
**Pull request by guyer**

### CODE REVIEW

File: steppyngstounes/parsimoniousStepper.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        You'll have to do some imports from numpy to get it to work.  It's
+        not pretty, but it was enough for my use case.  I'd be interested
+        to look at papers that discuss this problem if you have references
+        nearby.  Note that I also have a version that calls func on
+        multiple processes (using ipython).  With 2 quad cores, this sped
+        up considerably my application.
+
+        HTH,
+
+        David
```
Comment (huard) [inline steppyngstounes/parsimoniousStepper.py:37]:
```suggestion
    r"""Non-monotonic stepper that attempts to explore sparsely explored regions, with a preference for regions of high curvature.

```

**Review by huard — APPROVED** [review]

File: steppyngstounes/__init__.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 from .checkpointStepper import *
```
Comment (wd15) [inline steppyngstounes/__init__.py:2]: Are all these import *s necessary? Why not just import specific things?

**Review by wd15 — APPROVED** [review]

File: steppyngstounes/__init__.py (hunk 2 above)
Comment (wd15) [reply steppyngstounes/__init__.py:2]: It makes it more difficult for others reading the code to figure out where things come from.

File: steppyngstounes/parsimoniousStepper.py (hunk 1 above)
Comment (guyer) [reply steppyngstounes/parsimoniousStepper.py]: Much obliged, David. I hadn't intended to leave the email in, but included it for provenance during development.

File: steppyngstounes/__init__.py (hunk 2 above)
Comment (guyer) [reply steppyngstounes/__init__.py]: I was trying to be general (it's what FiPy and numpy do), but it just made everything opaque. I've switched to explicit imports.

**Review by tkphd — APPROVED** [review]
Looks great! Curious to try it with a simple system, to compare x, y, and L2 norm for the fixed & pærsymounïoſ steppers.

**Comment by guyer** [comment]
Rebased and merged in #2

**Pull request closed without merging**